### ISSUE TITLE
Error when making a shapes layer

### ISSUE THREAD

**Issue by psobolewskiPhD** [haesleinhuepf/napari-brightness-contrast#7]
I was playing with the plugin with an open image layer.
I set it the way I wanted and hid the plugin.
Then I clicked the Shapes layer Code which made this Error:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari/_qt/containers/_base_item_view.py in selectionChanged(self=, selected=, deselected=)
     76 
     77         if not self._root.selection.events.changed._emitting:
---> 78             self._root.selection.update(sel)
        self._root.selection.update = , })>
        sel = {}
     79             self._root.selection.difference_update(desel)
     80         return super().selectionChanged(selected, deselected)

~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari/utils/events/containers/_set.py in update(self=Selection({, }), others={})
    103             to_add = {self._pre_add_hook(i) for i in to_add}
    104             self._set.update(to_add)
--> 105             self._emit_change(added=set(to_add), removed={})
        self._emit_change = , })>
        global added = undefined
        global set = undefined
        to_add = {}
        global removed = undefined
    106 
    107     def copy(self) -> EventedSet[_T]:

~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari/utils/events/containers/_selection.py in _emit_change(self=Selection({, }), added={}, removed={})
     64     def _emit_change(self, added=set(), removed=set()):
     65         self._update_active()
---> 66         return super()._emit_change(added=added, removed=removed)
        global super._emit_change = undefined
        added = {}
        removed = {}
     67 
     68     def __repr__(self) -> str:

~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari/utils/events/containers/_set.py in _emit_change(self=Selection({, }), added={}, removed={})
     59     def _emit_change(self, added=set(), removed=set()):
     60         # provides a hook for subclasses to update internal state before emit
---> 61         self.events.changed(added=added, removed=removed)
        self.events.changed = 
        added = {}
        removed = {}
     62 
     63     def add(self, value: _T) -> None:

~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari/utils/events/event.py in __call__(self=, *args=(), **kwargs={'added': {}, 'removed': {}})
    713                     continue
    714 
--> 715                 self._invoke_callback(cb, event if pass_event else None)
        self._invoke_callback = >
        cb = >
        event = 
        pass_event = True
    716                 if event.blocked:
    717                     break

~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari/utils/events/event.py in _invoke_callback(self=, cb=>, event=)
    750                 self.disconnect(cb)
    751                 return
--> 752             _handle_exception(
        global _handle_exception = 
        self.ignore_callback_errors = False
        self.print_callback_errors = 'reminders'
        self = 
        global cb_event = undefined
        cb = >
        event = 
    753                 self.ignore_callback_errors,
    754                 self.print_callback_errors,

~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari/utils/events/event.py in _invoke_callback(self=, cb=>, event=)
    737         try:
    738             if event is not None:
--> 739                 cb(event)
        cb = >
        event = 
    740             else:
    741                 cb()

~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari_brightness_contrast/_dock_widget.py in _on_selection(self=, event=)
    116     def _on_selection(self, event):
    117         # redraw when layer selection has changed
--> 118         self.redraw()
        self.redraw = >
    119 
    120     def redraw(self, rebuild_gui=True):

~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari_brightness_contrast/_dock_widget.py in redraw(self=, rebuild_gui=True)
    185         selected_layers = self.selected_image_layers()
    186         for layer in self.viewer.layers:
--> 187             layer.events.contrast_limits.disconnect(self._data_changed_event)
        layer.events.contrast_limits.disconnect = undefined
        self._data_changed_event = >
    188             layer.events.data.disconnect(self._data_changed_event)
    189             if layer in selected_layers:

~/Dev/miniforge3/envs/napari-env/lib/python3.9/site-packages/napari/utils/events/event.py in __getattr__(self=, name='contrast_limits')
    927 
    928     def __getattr__(self, name) -> EventEmitter:
--> 929         return object.__getattribute__(self, name)
        global object.__getattribute__ = undefined
        self = 
        name = 'contrast_limits'
    930 
    931     def __getitem__(self, name: str) -> EventEmitter:

AttributeError: 'EmitterGroup' object has no attribute 'contrast_limits'
```

```

napari: 0.4.13rc0
Platform: macOS-12.1-arm64-arm-64bit
System: MacOS 12.1
Python: 3.9.9 | packaged by conda-forge | (main, Dec 20 2021, 02:41:07) [Clang 11.1.0 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.21.5
SciPy: 1.7.3
Dask: 2021.12.0
VisPy: 0.9.5.dev5

OpenGL:
- GL version: 2.1 Metal - 76.3
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1680x1050, scale 2.0

Plugins:
- aicsimageio-in-memory: 0.4.1
- aicsimageio-out-of-memory: 0.4.1
- clEsperanto: 0.12.0
- console: 0.0.4
- napari-brightness-contrast: 0.1.6
- napari-time-slicer: 0.4.2
- napari-tools-menu: 0.1.9
- napari_skimage_regionprops1: 0.2.7
- napari_skimage_regionprops2: 0.2.7
- ome-types: 0.2.10
- scikit-image: 0.4.13rc0
- svg: 0.1.5

```

**Comment by haesleinhuepf** [comment]
Hi Peter @psobolewskiPhD ,

thanks for reporting! Would you be interested in helping to fix the issue? Presumably it is just adding a `if hasattr(layer.events, 'contrast_limits'):` above this line:
https://github.com/haesleinhuepf/napari-brightness-contrast/blob/main/napari_brightness_contrast/_dock_widget.py#L187

**Comment by psobolewskiPhD** [comment]
Hi @haesleinhuepf I forgot about this and also didn't have time, but I tried this and it worked. I have a branch with the fix. Also wanted to write a test and had some issues with tests, so I have a branch that makes tests work (add pytest, pytest-qt to setup.cfg) and adds 2 tests for the current release:
https://github.com/psobolewskiPhD/napari-brightness-contrast/tree/fix_tests

Should I merge them into 1 PR? or 2 separate ones?

Edit: Also I forgot my VS Code runs `black` and some other linting stuff, so the 1 line edit now also has a buncha formatting changes. Do you want me to try and revert and just make it the one line?

**Comment by haesleinhuepf** [comment]
Hi @psobolewskiPhD 

let's keep this two separate discussions. If you could send two PRs, that would be great.

Thanks for working on this!

Robert

**Comment by haesleinhuepf** [comment]
fixed in #9
